src/utils/viewer/viewer.cpp

```diff
@@ -352,14 +406,32 @@ void Viewer::drawEvent() {
     physicsManager_->debugDraw(projM * camM);
   }
 
+  textShader_
+      .setTransformationProjectionMatrix(
+          textProjection_ *
+          Mn::Matrix3::translation(
+              1.0f / textProjection_.rotationScaling().diagonal()))
+      .setColor(Color3{1.0f})
+      .bindVectorTexture(cache_.texture());
+  performanceText_->mesh().draw(textShader_);
+
   swapBuffers();
-  timeline_.nextFrame();
   redraw();
+
+  totalFrames_++;
+  if ((totalFrames_ & fpsUpdateFrequencyMask) == 0) {
```

Comment:
Why not 
```suggestion
  if ((totalFrames_ % fpsUpdateFrequencyMask) == 0) {
```

Reply:
Because it is a bit faster.

Reply:
I don't think that matters in this context. In a tight loop over thousands of elements probably yes, but here any additional latency from the modulo op is negligible compared to function call overhead, VSync waiting and such. I'd vote for having the modulo and not requiring this to be `2^n - 1`, as that's simpler.

Reply:
I learned it a couple of years ago when implementing the lock-less ring buffer. It is a bit faster and will benefit e.g., a server program which needs to run for long time (no need to be in a tight loop). It comes natural to me, so I did not think too much about it when I wrote the code.

If you guys do not like it, I will change it using modulo. This program is not that time-critical after all. I agree.